library/std/src/net/tcp/tests.rs

```diff
@@ -46,6 +46,37 @@ fn connect_error() {
     }
 }
 
+#[test]
+fn connect_timeout_to_unreachable_address() {
+    let now = Instant::now();
+    match TcpStream::connect_timeout(&format!("1.1.1.1:9999").parse().unwrap(), Duration::MAX) {
+        Ok(..) => panic!("connected to an unreachable address, this is impossible"),
+        Err(e) => assert_eq!(e.kind(), ErrorKind::TimedOut),
+    }
+
+    assert!(now.elapsed() > Duration::from_secs(20));
+}
+
+#[test]
```

Comment:
```suggestion
#[test]
#[cfg_attr(target_env = "sgx", ignore)] // FIXME: https://github.com/fortanix/rust-sgx/issues/31
```

Unfortunately the SGX platform does not currently support `connect_timeout`.